libvast/src/address.cpp

```diff
@@ -16,13 +16,104 @@
 
 #include <arpa/inet.h>
 #include <netinet/in.h>
+#include <openssl/crypto.h>
+#include <openssl/err.h>
+#include <openssl/evp.h>
+#include <openssl/ossl_typ.h>
 #include <sys/socket.h>
 
 #include <cstdlib>
 #include <utility>
 
 namespace vast {
 
+namespace {
+
+inline uint32_t bitmask32(size_t bottom_bits) {
+  return bottom_bits >= 32 ? 0xffffffff : ((uint32_t{1} << bottom_bits) - 1);
+}
+
+class address_encryptor {
+public:
+  address_encryptor(const std::array<address::byte_type, 32>& key) {
```

Comment:
Single-argument constructors should be marked explicit. If you want to make a constructor implicit intentionally, use `explicit(false)`.